acceptance-tests/src/test-support/java/tech/pegasys/artemis/test/acceptance/dsl/ArtemisNode.java

```diff
@@ -137,39 +138,70 @@ public UnsignedLong getGenesisTime() throws IOException {
     return fetchGenesisTime();
   }
 
-  public void waitForNewBlock() throws IOException {
-    final Bytes32 startingBlockRoot = getCurrentBeaconHead().getBlockRoot();
-    waitFor(
-        () -> assertThat(getCurrentBeaconHead().getBlockRoot()).isNotEqualTo(startingBlockRoot));
+  public void waitForNewBlock() {
+    final Bytes32 startingBlockRoot = waitForBeaconHead().getBlockRoot();
+    waitFor(() -> assertThat(waitForBeaconHead().getBlockRoot()).isNotEqualTo(startingBlockRoot));
   }
 
-  public void waitForNewFinalization() throws IOException {
-    UnsignedLong startingFinalizedEpoch = getChainHead().finalized_epoch;
+  public void waitForNewFinalization() {
+    UnsignedLong startingFinalizedEpoch = waitForChainHead().finalized_epoch;
     LOG.debug("Wait for finalized block");
     waitFor(
-        () -> assertThat(getChainHead().finalized_epoch).isNotEqualTo(startingFinalizedEpoch), 540);
+        () -> assertThat(waitForChainHead().finalized_epoch).isNotEqualTo(startingFinalizedEpoch),
+        540);
   }
 
   public void waitUntilInSyncWith(final ArtemisNode targetNode) {
     LOG.debug("Wait for {} to sync to {}", nodeAlias, targetNode.nodeAlias);
+    waitFor(() -> assertThat(waitForBeaconHead()).isEqualTo(targetNode.waitForBeaconHead()), 300);
```

Comment:
We should probably be using `fetchBeaconHead` here rather than `waitForBeaconHead`. Otherwise we're effectively nesting a waiter inside a waiter.  It will work but failure times become a lot longer unnecessarily.